src/controllers/commands/CommandController.cpp

```diff
@@ -2829,6 +2829,54 @@ void CommandController::initialize(Settings &, Paths &paths)
         return "";
     });
 
+    this->registerCommand("/banid", [formatBanTimeoutError](
+                                        const QStringList &words,
+                                        auto channel) {
+        auto *twitchChannel = dynamic_cast<TwitchChannel *>(channel.get());
+        if (twitchChannel == nullptr)
+        {
+            channel->addMessage(makeSystemMessage(
+                QString("The /banid command only works in Twitch channels")));
+            return "";
+        }
+
+        const auto *usageStr =
+            "Usage: \"/banid <userid> [reason]\" - Permanently prevent a user "
+            "ID from chatting. Reason is optional and will be shown to the "
+            "target user and other moderators. Use \"/unban <username>\" to "
+            "remove a ban.";
+        if (words.size() < 2)
+        {
+            channel->addMessage(makeSystemMessage(usageStr));
+            return "";
+        }
+
+        auto currentUser = getApp()->accounts->twitch.getCurrent();
+        if (currentUser->isAnon())
+        {
+            channel->addMessage(
+                makeSystemMessage("You must be logged in to ban someone!"));
+            return "";
+        }
+
+        auto target = words.at(1);
+        auto reason = words.mid(2).join(' ');
+
+        getHelix()->banUser(
+            twitchChannel->roomId(), currentUser->getUserId(), target,
+            boost::none, reason,
+            [] {
+                // No response for bans, they're emitted over pubsub/IRC instead
+            },
+            [channel, target, formatBanTimeoutError](auto error, auto message) {
+                auto errorMessage =
+                    formatBanTimeoutError("ban", error, message, "#" + target);
+                channel->addMessage(makeSystemMessage(errorMessage));
```

Comment:
Unfortunately, Twitch doesn't send us an informative error message when a user tries to pass a login name instead of ID to the command:

```json
{
  "data": [],
  "error": "Internal Server Error",
  "status": 500
}
```

So this results in: `Failed to ban user - An unknown error has occurred.`

Reply:
I'm not _against_ validating this against a `\d+` regex as I'm doubtful Twitch will ever send a correct error for this, but we could also just do nothing

Reply:
I feel like this is a bit of an advanced command, it shouldn't be necessary to put too much effort into parsing weird user errors.